blockchain/types/transaction.go

```diff
@@ -251,7 +251,8 @@ func (tx *Transaction) ValidateMutableValue(db StateDB, signer Signer, currentBl
 		}
 	} else {
 		pubkey, err := SenderPubkey(signer, tx)
-		if err != nil || accountkey.ValidateAccountKey(tx.validatedSender, accKey, pubkey, tx.GetRoleTypeForValidation()) != nil {
+		validateAccountKeyErr, _ := accountkey.ValidateAccountKey(tx.validatedSender, accKey, pubkey, tx.GetRoleTypeForValidation(), currentBlockNumber)
+		if err != nil || validateAccountKeyErr != nil {
```

Comment:
I think this kind of error checking is dangerous. If `SenderPubkey` returns an error, `pubkey` will be `nil` and this value may cause a panic with "nil pointer dereference error" inside of `accountkey.ValidateAccountKey`.  Currently, `accountkey.ValidateAccountKey` seems safe for the `nil` `pubkey`, but may be a developer change the code later without consideration.  Could you check `err` first, and then call `accountkey.ValidateAccountKey`? 
If you agreed with my opinion, please change other similar codes on this PR also.

Reply:
@aidan-kwon Oh.. It's really nice comment.. You're right. However, as we discussed in offline, `Validate` doesn't need to return second return value, so I reverted the code.
By reverting the code, `validateAccountKey` will not be executed when pubKey is nil.